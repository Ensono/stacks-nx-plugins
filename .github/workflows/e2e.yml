name: e2e

on:
    push:
        branches:
            - main
    pull_request:

permissions:
    actions: read
    contents: read

jobs:
    main:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v4
              with:
                  filter: tree:0
                  fetch-depth: 0
            
            - uses: pnpm/action-setup@v4
              with:
                  version: '10'

            # This enables task distribution via Nx Cloud
            # Run this command as early as possible, before dependencies are installed
            # Learn more at https://nx.dev/ci/reference/nx-cloud-cli#npx-nxcloud-startcirun
            # Connect your workspace by running "nx connect" and uncomment this line to enable task distribution
            # - run: pnpm dlx nx-cloud start-ci-run --distribute-on="3 linux-medium-js" --stop-agents-after="e2e-ci"
            #   env:
            #       NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
                  
            # Cache node_modules
            - uses: actions/setup-node@v4
              with:
                  node-version-file: '.nvmrc'
                  cache: 'pnpm'

            - run: pnpm install --frozen-lockfile
            - uses: nrwl/nx-set-shas@v4

            # Prepend any command with "nx-cloud record --" to record its logs to Nx Cloud
            # - run: npx nx-cloud record -- echo Hello World
            # Nx Affected runs only tasks affected by the changes in this PR/commit. Learn more: https://nx.dev/ci/features/affected
            # When you enable task distribution, run the e2e-ci task instead of e2e
            - run: pnpm exec nx affected -t lint build

            - run: pnpm exec nx affected -t test --runInBand

            - run: pnpm exec nx affected -t e2e --parallel=1 --runInBand --verbose
