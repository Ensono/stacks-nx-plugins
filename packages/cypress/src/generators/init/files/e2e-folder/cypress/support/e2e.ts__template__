// ***********************************************************
// This example support/index.js is processed and
// loaded automatically before your test files.
//
// This is a great place to put global configuration and
// behavior that modifies Cypress.
//
// You can change the location of this file or turn off
// automatically serving support files with the
// 'supportFile' configuration option.
//
// You can read more here:
// https://on.cypress.io/configuration
// ***********************************************************
/* eslint-disable unicorn/prevent-abbreviations */

// Import commands.js using ES2015 syntax:
import './commands';
import registerCypressGrep from '@cypress/grep/src/support';
import addContext from 'mochawesome/addContext';

registerCypressGrep();

// Append screenshots on failure to test reports
Cypress.on('test:after:run', (test, runnable) => {
  if (test.state === 'failed') {
    const screenshot = `${Cypress.config('screenshotsFolder')}/${
      Cypress.spec.name
    }/${runnable.parent?.title} -- ${test.title} (failed).png`;
    // @ts-ignore
    addContext({ test }, screenshot);
  }
});