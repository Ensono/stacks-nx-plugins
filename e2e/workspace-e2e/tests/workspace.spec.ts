import {
    checkFilesExist,
    readJson,
    runNxCommandAsync,
} from '@nrwl/nx-plugin/testing';
import { newProject, cleanup } from '@ensono-stacks/e2e';
describe('workspace', () => {
    beforeAll(async () => {
        await newProject(['@ensono-stacks/workspace']);
    });

    afterAll(() => {
        runNxCommandAsync('reset');
        cleanup();
    });

    it('runs the install generator', async () => {
        await runNxCommandAsync(
            `generate @ensono-stacks/workspace:install --no-interactive`,
            {
                env: {
                    // Nx create will not create a git context for e2e,
                    // so skip husky installation
                    // https://github.com/typicode/husky/blob/main/src/index.ts#L14
                    HUSKY: '0',
                },
            },
        );

        expect(() =>
            checkFilesExist(
                '.eslintrc.json',
                '.husky/commit-msg',
                '.husky/pre-commit',
                '.husky/prepare-commit-msg',
            ),
        ).not.toThrow();

        const packageJson = readJson('package.json');

        expect(packageJson).toMatchObject(
            expect.objectContaining({
                config: {
                    commitizen: {
                        path: '@commitlint/cz-commitlint',
                    },
                },
            }),
        );
    }, 100000);
});
