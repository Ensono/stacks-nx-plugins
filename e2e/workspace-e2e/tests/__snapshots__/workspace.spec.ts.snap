// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workspace updates the eslintrc.json 1`] = `
{
  "extends": [
    "airbnb/base",
    "plugin:prettier/recommended",
    "plugin:unicorn/recommended",
    "plugin:compat/recommended",
    "plugin:import/recommended",
    "plugin:import/typescript",
    "plugin:security/recommended-legacy",
    "plugin:jsx-a11y/recommended",
    "plugin:jest/recommended",
    "plugin:jest-dom/recommended",
    "plugin:no-unsanitized/DOM",
  ],
  "ignorePatterns": [
    "**/*",
  ],
  "overrides": [
    {
      "files": [
        "*.ts",
        "*.tsx",
        "*.js",
        "*.jsx",
      ],
      "rules": {
        "@nx/enforce-module-boundaries": [
          "error",
          {
            "allow": [],
            "depConstraints": [
              {
                "onlyDependOnLibsWithTags": [
                  "*",
                ],
                "sourceTag": "*",
              },
            ],
            "enforceBuildableLibDependency": true,
          },
        ],
        "import/extensions": [
          "error",
          "always",
          {
            "js": "never",
            "jsx": "never",
            "ts": "never",
            "tsx": "never",
          },
        ],
        "import/no-extraneous-dependencies": "off",
        "import/order": [
          "error",
          {
            "alphabetize": {
              "order": "asc",
            },
            "groups": [
              [
                "builtin",
                "external",
              ],
              [
                "internal",
              ],
              [
                "parent",
                "sibling",
                "index",
              ],
            ],
            "newlines-between": "always",
            "pathGroups": [
              {
                "group": "internal",
                "pattern": "@proj/**",
              },
            ],
          },
        ],
        "import/prefer-default-export": "off",
        "jest/expect-expect": [
          "error",
          {
            "assertFunctionNames": [
              "expect",
              "cy.**.should",
              "**.contains",
            ],
          },
        ],
        "no-console": "off",
        "no-duplicate-imports": "error",
        "no-irregular-whitespace": "error",
        "no-restricted-exports": "off",
        "no-shadow": "off",
        "no-use-before-define": "off",
        "security/detect-object-injection": "off",
        "unicorn/filename-case": "off",
        "unicorn/no-array-for-each": "off",
        "unicorn/no-array-reduce": "off",
        "unicorn/no-for-loop": "off",
        "unicorn/no-null": "off",
        "unicorn/prefer-module": "off",
        "unicorn/prevent-abbreviations": [
          "error",
          {
            "allowList": {
              "getServerSideProps": true,
              "getStaticProps": true,
            },
            "replacements": {
              "env": {
                "environment": false,
              },
            },
          },
        ],
      },
    },
    {
      "extends": [
        "plugin:@nx/typescript",
      ],
      "files": [
        "*.ts",
        "*.tsx",
      ],
      "rules": {
        "@typescript-eslint/ban-ts-comment": "off",
        "@typescript-eslint/explicit-function-return-type": "off",
        "@typescript-eslint/interface-name-prefix": "off",
        "@typescript-eslint/no-empty-function": "off",
        "@typescript-eslint/no-explicit-any": "off",
        "@typescript-eslint/no-shadow": "error",
        "@typescript-eslint/no-unsafe-assignment": "off",
        "@typescript-eslint/no-unsafe-call": "off",
        "@typescript-eslint/no-unsafe-member-access": "off",
        "@typescript-eslint/no-unused-vars": "off",
        "@typescript-eslint/no-use-before-define": "error",
        "unicorn/prefer-node-protocol": "off",
      },
    },
    {
      "files": "*.json",
      "parser": "jsonc-eslint-parser",
      "rules": {
        "unicorn/prevent-abbreviations": "off",
      },
    },
    {
      "files": "jest.config.ts",
      "rules": {
        "unicorn/no-abusive-eslint-disable": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "@typescript-eslint",
    "@nx",
    "import",
    "security",
    "jsx-a11y",
    "jest",
  ],
  "root": true,
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": "tsconfig.base.json",
      },
    },
  },
}
`;
